pkg/test/utils/kubernetes.go

```diff
@@ -573,6 +573,11 @@ func InstallManifests(ctx context.Context, client client.Client, dClient discove
 
 		for _, obj := range objs {
 			if len(kinds) > 0 && !MatchesKind(obj, kinds...) {
+				var expectedKinds []string
+				for _, k := range kinds {
```

Comment:
this `for` loop is in a `for` loop and does not need to be... it is only creating `expectedKinds` for the log message after the `if` match... lets move it prior to the outer loop.

Otherwise... super smart!  thanks!

Reply:
yeah I did not want to allocate that array if all kinds matches, which should be majority of cases

Reply:
hmm... that is a good point.. that intention would benefit from a comment... I'll revert.

Reply:
well... it could be viewed a number ways... (can't guess how this function might be used in the future).. currently it is called most of the time with 0 or 1 kinds... and with the merge of the CRDv1 PR it will be 0 or 2 kinds.   It is also only called during setup... It seems more readable to pull it out of the loop... wdyt?

Reply:
my preference would still be to have it in the loop, it's closer to the usage so it's actually better for me even from readability perspective.

but I don't feel strongly about neither so whatever works for you better.

Reply:
didn't feel strongly about it... I'll revert (to your PR:))...  committing on another's PR is cool... until it isn't :) 
it is clear what the intention is... I'll add a comment and revert to your code structure.